src/components/HomeBlocks/components/LeaderboardHomeBlockCreatorItem.tsx

```diff
@@ -46,19 +68,40 @@ export const LeaderHomeBlockCreatorItem = ({
                   textSize="lg"
                   size="md"
                 />
-                <Stack spacing={4}>
-                  <Text>{user.username}</Text>
+                <Stack spacing={4} style={{ overflow: 'hidden' }}>
+                  <Text
+                    style={{
+                      whiteSpace: 'nowrap',
+                      overflow: 'hidden',
+                      textOverflow: 'ellipsis',
+                    }}
+                  >
+                    {user.username}
+                  </Text>
                   <Group spacing={4}>
                     <IconTrophy size={12} />
-                    <Text size="xs">{numberWithCommas(score)}</Text>
+                    <Text size="xs">{numberWithCommas(score) || 0}</Text>
```

Comment:
Let's use this handy helper method
```suggestion
                    <Text size="xs">{abbreviateNumber(score)}</Text>
```